### ISSUE TITLE
Unable to sign with Jade in Sparrow 1.7.8

### ISSUE THREAD

**Issue by rserranon** [sparrowwallet/sparrow#1047]
I did a previous test with version 1.7.7 and I was able to sign a multisig Tx with Jade.  

In this last version, I get the psbt loaded in the Jade, I'm able to verify the amount and fee and sign it, but when I scan the QR code animation in Sparrow, it finishes the scan, but the Tx does not get signed.

Mac OS Ventura 13.2.1 M1
Jade firmware: 0.1.48

**Comment by craigraw** [comment]
Are you sure it is not a passphrase issue on the Jade?

You can also try setting `useZbar` to `false` in the Sparrow config file (close Sparrow first) but I doubt this is the issue.

**Comment by shaun-widmer** [comment]
Hey guys, I'm having the same issue with my blockstream jade not signing multi-sig transactions. I'm also using Sparrow 1.7.8. (I'm attempting to sign with the USB cable, not QR)

I even purchased another blockstream jade device and the new one doesn't work either!

Every part of the signing process works on the Jade (veryfying the amount for sending, the fees, etc...)  until the very last step in the signing process. It just says "Processing..." on the Jade device but never finished signing the transaction. 

I created a single sig wallet on Sparrow to test signing with the same jade device and it works flawlessly... but in the multi sig it doesn't work. 

I greatly appreciate any help!

Setup:

- I have a multi-sig setup with a blockstream Jade, a ledger and a trezor. Every wallet works to sign transactions except the Jade. 

- I have the latest version of firmware installed on the Jade wallet. 

- I used blockstream green to initialize the jade, but I'm NOT running blockstream green or any other wallet application while signing with the sparrow wallet.

**Comment by shaun-widmer** [comment]
The passphrase is definitely not the problem for me

**Comment by craigraw** [comment]
I have just signed a test multisig transaction with my Jade using both USB and QR, so it is definitely possible (also MacOS on M1). 

@shaun-widmer let's keep your issue (which I believe is separate to the issue you have created.

@rserranon Can you confirm whether a passphrase was used during import or signing with your Jade? It appears that the Jade returned an invalid signature.

**Comment by rserranon** [comment]
> I have just signed a test multisig transaction with my Jade using both USB and QR, so it is definitely possible (also MacOS on M1). 
> 
> 
> 
> @shaun-widmer let's keep your issue (which I believe is separate) to the issue you have created.
> 
> 
> 
> @rserranon Can you confirm whether a passphrase was used during import or signing with your Jade? It appears that the Jade returned an invalid signature.

Thx Craig,

I can confirm I'm not using a passphrase on my Jade.

**Comment by craigraw** [comment]
I suggest trying to create a transaction with a single input UTXO and sign that to reduce the variables. You don't need to broadcast it.

**Comment by moneymanolis** [comment]
@shaun-widmer which firmware version of the Jade are you using? Try with Jade firmware: 0.1.48 or higher.

**Comment by Bruno-N** [comment]
Hi,

I'm having the same issue using Jade, creating the initial transaction with one signature from the first device works, but trying to add a second signature with another device doesn't, after going through the prompt on the device Sparrow returns to the transaction screen and nothing happens.

I also tried the same process on a transaction with only 1 output and got the same issue.

Running on Windows.

**Comment by craigraw** [comment]
I've tried to reproduce this using firmware 0.1.42 and now 1.0.21 without success - my Jade produces a valid signature for a multisig transaction. This works using both USB and QR.

One way (and probably the only way) this behaviour could happen is if the Jade returned the input PSBT without adding any signatures to it. If the Jade added an invalid signature (or returned an invalid PSBT), Sparrow would show an error dialog. If a different PSBT was returned, a new transaction tab would open. Simply returning the same PSBT results in the two PSBTs being combined, which means no effective change. I can't be completely sure without reproducing the behaviour, but as far as I can tell this is the only way that this can happen.

cc @JamieDriver

**Comment by Bruno-N** [comment]
Hi Craig,

I figured it out, I believe it is a "workflow" issue, and it might be the same for the other people commenting.

I created a multisig wallet using one hardware wallet and 2 XPubs, then I created a transaction from this wallet, signed and saved it. I then proceeded to open the PSBT file on another computer that had the same multisig wallet, but using another hardware wallet and the other XPubs, then the error happens when I load the previously signed PSBT and try to add the second signature. 

If I create the transaction from a wallet that has the 3 XPubs, and then load it on the first wallet, sign it, then load that onto the second wallet and sign it, it works.

Basically the issue comes from creating the "initial transaction" in the wallet with the "1 HW and 2 XPubs" setup.

I hope this helps you to replicate the issue.

**Comment by craigraw** [comment]
Thanks, that helps.

Quick question - are you copying the master fingerprints and derivation paths across when you copy the xpubs? Put another way, are the output descriptors for all the wallets identical?

**Comment by Bruno-N** [comment]
I'm using the same derivation path, but I'm not touching the master fingerprint when setting up, to be fair I don't know what that is for.

I just manually set the label, derivation path (m/48'/0'/0'/2') and the XPub.

**Comment by JamieDriver** [comment]
Jade does indeed use fingerprints to find the inputs it may need to sign (it double checks the full pubkeys in case of fingerprint collision - but fingerprint match is the first 'filter' before it even thinks about deriving any pubkeys).

**Comment by Bruno-N** [comment]
Thanks guys,

That was indeed the problem, I was incorrectly assuming the fingerprint wouldn't matter.

It would be nice if it was automatically populated when you manually input the XPub.

**Comment by craigraw** [comment]
> It would be nice if it was automatically populated when you manually input the XPub.

It is not derived from the xpub - it is in fact defined as the first four bytes of the master public key derived from the seed. 

The simplest way to create your distributed setup is to use the BSMS import/export option, or simply copy the output descriptor across (see the Edit button on the Setttings tab). This will setup the wallet and copy the master fingerprints, derivation paths and xpubs for all the keystores. You can then replace the keystore that you use to sign by re-importing the Jade there.

@rserranon Can you confirm that we have identified the cause of this issue for you as well?

**Comment by rserranon** [comment]
@craigraw I'm traveling and I don't have access to my Jade until September 15th.

**Comment by craigraw** [comment]
Closing this off - assume the issue has been resolved.
